### PR TITLE
DG-25817 Adding transfer site ownership API to sdk

### PR DESCRIPTION
**Pull request by devanshu2316**

### CODE REVIEW

File: lib/sfrest/site_ownership.rb [hunk 1]
```diff
@@ -37,5 +37,19 @@ def make_default_owner(username)
     def remove_default_owner
       @conn.delete('/api/v1/site-ownership')
     end
+
+    # Transfers site ownership.
+    # @param [Integer] site id.
+    # @param [String] username.
+    # @param [String] email.
```
Comment (glitchinfinity) [inline lib/sfrest/site_ownership.rb:44]: Is this parameter missing from function definition?

Comment (devanshu2316) [reply lib/sfrest/site_ownership.rb:44]: We need only one parameter out of two, either username or email.

Comment (glitchinfinity) [reply lib/sfrest/site_ownership.rb:44]: Then we should specify that in the description and add the necessary logic to send the `email` parameter as well. Currently this only works for the username usecase. There's no way to use it using the email usecase.
Also, we should extend the rspec to test both the username and email usecases.

Comment (devanshu2316) [reply lib/sfrest/site_ownership.rb]: Fixed :)

**Pull request merged**